paimon-core/src/main/java/org/apache/paimon/mergetree/MergeTreeWriter.java

```diff
@@ -106,8 +108,8 @@ public MergeTreeWriter(
         this.sortMaxFan = sortMaxFan;
         this.sortCompression = sortCompression;
         this.ioManager = ioManager;
-        this.keyType = writerFactory.keyType();
-        this.valueType = writerFactory.valueType();
+        this.keyType = keyType;
+        this.valueType = valueType;
```

Comment:
Why do you change this? `writerFactory.keyType()` and `writerFactory.valueType()` are not changed,